pkg/sql/schemachanger/scbuild/internal/scbuildstmt/configure_zone.go

```diff
@@ -504,17 +706,41 @@ func lookUpSystemZonesTable(
 				if e.DatabaseID == targetID {
 					zone = e.ZoneConfig
 					seqNum = e.SeqNum
+					// TODO(before merge) is this a small optimization we can do? DatabaseZoneConfig is on the
```

Comment:
adding support for sequences and mat views rn (not too bad -- just have to do some more logic since it'll be broken up into 3 seperate elements)

also the answer to my question here: the answer is no since that's not how ForEach works -- i might be interested in making some sort of `func (c *ElementCollection[E]) Find(pred func(e E) bool) E` tho so we can short-circuit?